lib/pf/Authentication/Source/LDAPSource.pm

```diff
@@ -224,6 +225,10 @@ sub _connect {
   my $logger = Log::Log4perl::get_logger(__PACKAGE__);
   my ($LDAPServer, $LDAPServerPort);
   my @LDAPServers = split(/\s*,\s*/, $self->{'host'});
+  
+  # Lookup the server hostnames to IPs so they can be shuffled better and to improve the failure detection
+  @LDAPServers = map { valid_ip($_) ? $_ : @{resolve($_)} } @LDAPServers;
```

Comment:
This does not handle the case where a port is defined in the host list.

Reply:
I just figured out we can ignore the override port specification for hostnames (it still works for IPs)

That feature is undocumented and not used by any of our customers (makes no sense to use different ports per LDAP hosts). This just looks like a leftover of PF version 3 when authentication sources were Perl modules

Reply:
That feature was added in this commit 4d1c2db2315ca8d084bba3172c3bd3dc592ff6b6 (This is a 4.0.0).
This was a feature added since we added support for multiple LDAP servers.

My thoughts are this if it is going backported do minimal things that can break a feature that might be used by someone. If this is going to be backported support the override port.

If not then to keep the code clean remove the splitting of the host/port below and add a message in the upgrade notes about removing the feature.

Also add upgrade script should be added to clean up and remove the port from the hostnames.

Reply:
Lets just skip the backporting and manually patch whoever needs this (shouldn't be too many people)

I'll remove the code of multiple ports and add a note in the guide